script/releases/v0.1-eigenpod-example/1-eoa.s.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "zeus-templates/templates/EOADeployer.sol";
+
+import "src/contracts/pods/EigenPod.sol";
+import "src/contracts/pods/EigenPodManager.sol";
+import "src/contracts/interfaces/IEigenPodManager.sol";
+
+import "zeus-templates/utils/StringUtils.sol";
+
+contract DeployEigenPodAndManager is EOADeployer {
+    using StringUtils for string;
+
+    /// @notice deploys an EigenPod and returns the deployed address
+    function _deploy(Addresses memory, Environment memory, Params memory params) internal override returns (Deployment[] memory) {
+
+        vm.startBroadcast();
+
+        // create a defunct EigenPodManager
+        address newEigenPodManager = address(new EigenPodManager(
+            IETHPOSDeposit(address(1)),
+            IBeacon(address(2)),
+            IStrategyManager(address(3)),
+            ISlasher(address(4)),
+            IDelegationManager(address(5))
+        ));
+
+        // record new deployment
+        _deployments.push(Deployment({
+            name: type(EigenPodManager).name,
+            deployedTo: newEigenPodManager,
+            envToUpdate: "eigenPodManager.pendingImpl"
+        }));
+
+        // create and record new EigenPod pointing to defunct EigenPodManager
+        _deployments.push(Deployment({
+            name: type(EigenPod).name,
+            deployedTo: address(new EigenPod(
+                IETHPOSDeposit(params.ethPOS),
+                IEigenPodManager(newEigenPodManager), // update EigenPodManager address
+                params.EIGENPOD_GENESIS_TIME
+            )),
+            envToUpdate: "eigenPod.pendingImpl"
+        }));
+
+        vm.stopBroadcast();
+
+        return _deployments;
```

Comment:
Do we need to return this if it's in a state variable the parent method has access to?

Reply:
1) I prefer explicitly returning up values rather than implicitly modifying state variables.
2) I actually meant to have this not be in the parent contract 😅 and instead have every contract inheriting `EOADeployer` define their own `_deployments` array local to their own contract to avoid 1). 

Will update this in the templates branch and this PR